sdk/compiler/daml-lf-proto-decode/test/DA/Daml/LF/Proto3/DecodeTest.hs

```diff
@@ -0,0 +1,97 @@
+-- Copyright (c) 2025 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module DA.Daml.LF.Proto3.DecodeTest (
+        module DA.Daml.LF.Proto3.DecodeTest
+) where
+
+
+import           Data.Int
+import qualified Data.Vector                              as V
+
+import           DA.Daml.LF.Proto3.DecodeV2
+
+import           DA.Daml.LF.Ast
+import qualified Com.Digitalasset.Daml.Lf.Archive.DamlLf2 as P
+
+import           Test.Tasty.HUnit
+import           Test.Tasty
+
+entry :: IO ()
+entry = defaultMain $ testGroup "All tests" [ dec_tests ]
+
+------------------------------------------------------------------------
+-- Params
+------------------------------------------------------------------------
+testVersion :: Version
+testVersion = Version V2 PointDev
+
+------------------------------------------------------------------------
+-- DecodeTests
+------------------------------------------------------------------------
+dec_tests :: TestTree
+dec_tests = testGroup "decoding tests"
+  [ dec_pure_tests
+  , dec_interning_tests
+  ]
+
+emptyDecodeEnv :: DecodeEnv
+emptyDecodeEnv = DecodeEnv V.empty V.empty V.empty V.empty SelfPackageId testVersion
+
+decodeKindAssert :: P.Kind -> Kind -> Assertion
+decodeKindAssert pk k =
+  either
+    (\err -> assertFailure $ "Unexpected error: " ++ show err)
+    (\k' -> k @=? k')
+    (runDecode env (decodeKind pk))
+    where
+      env = emptyDecodeEnv
+
+
+decodeKindTest :: String -> P.Kind -> Kind -> TestTree
+decodeKindTest str pk k = testCase str $ decodeKindAssert pk k
+
+dec_pure_tests :: TestTree
```

Comment:
I know we don't have a style guide currently per-se, but I think you're emulating the GHC repo's "snake_case-means-internal, camelCase means public" API style here (correct me if I'm wrong). Since the rest of the codebase doesn't use this convention, I'm really not inclined to start it here. (I'll admit that I'm a bit biased since I never really liked the convention myself :P).